kernel/src/hil/touch.rs

```diff
@@ -0,0 +1,98 @@
+//! Interface for touch input devices
+
+use crate::ReturnCode;
+
+/// Touch Event Status
+#[derive(Debug, Copy, Clone)]
+pub enum TouchStatus {
+    Pressed,
+    Released,
+}
+
+/// Gesture event
+#[derive(Debug, Copy, Clone)]
+pub enum GestureEvent {
+    SwipeUp,
+    SwipeDown,
+    SwipeLeft,
+    SwipeRight,
+    ZoomIn,
+    ZoomOut,
+}
+
+/// A single touch event's data
+#[derive(Copy, Clone)]
+pub struct TouchEvent {
+    pub status: TouchStatus,
+    /// touch (x, y) position
+    pub x: usize,
+    pub y: usize,
+
+    /// touch id, value defined by the driver
+    pub id: usize,
+
+    /// touch area
+    pub area: Option<usize>,
```

Comment:
```suggestion
    /// Optional scaled value for the size of the touch. A larger value
    /// corresponds to a "fatter" touch. The size values range from 0
    /// to 65535.
    ///
    /// If a touchscreen does not provide information about the size of the touch,
    /// this must be set to `None`.
    pub size: Option<usize>,
```

I'm copying from the android API largely.